aws-s3-private-bucket/main.tf

```diff
@@ -0,0 +1,66 @@
+locals {
+  tags = {
+    project   = "${var.project}"
+    env       = "${var.env}"
+    service   = "${var.service}"
+    owner     = "${var.owner}"
+    managedBy = "terraform"
+  }
+}
+
+resource "aws_s3_bucket" "bucket" {
+  bucket = "${var.bucket_name}"
+  acl    = "private"
+
+  policy = "${data.aws_iam_policy_document.bucket_policy.json}"
+
+  versioning {
+    enabled = true
+  }
+
+  # TODO
+  #   logging {
+  #   target_bucket = ""
+  #   target_prefix = ""
+  # }
+
+  server_side_encryption_configuration {
+    rule {
+      apply_server_side_encryption_by_default {
+        sse_algorithm = "AES256"
+      }
+    }
+  }
+  tags = "${local.tags}"
+}
+
+resource "aws_s3_bucket_public_access_block" "bucket" {
+  bucket = "${aws_s3_bucket.bucket}"
```

Comment:
you might have to do aws_s3_bucket.bucket.id https://www.terraform.io/docs/providers/aws/r/s3_bucket.html#id-2